app/src/main/java/com/example/dispatcher/BaseFragment.kt

```diff
@@ -0,0 +1,31 @@
+package com.example.dispatcher
+
+import android.os.Bundle
+import android.view.View
+import androidx.fragment.app.Fragment
+import com.google.gson.Gson
+import com.google.gson.reflect.TypeToken
+import java.io.InputStreamReader
+
+abstract class BaseFragment : Fragment() {
+
+    protected val articleList: ArrayList<Article> = ArrayList()
+
+    protected open fun initArticleList() {
+        val jsonFileInputStream = resources.openRawResource(R.raw.articles)
+        val reader = InputStreamReader(jsonFileInputStream)
```

Comment:
this part of getting the data from a json file should be in a separate file. a manager.